code/modules/mob/living/living.dm

```diff
@@ -1333,16 +1333,37 @@
 			return//dont open the mobs inventory if you are picking them up
 	. = ..()
 
-/mob/living/proc/mob_pickup(mob/living/L)
+/**
+  * Creates a mob_holder object and puts the mob in it.
+  *
+  * Usually followed up by putting the new holder in something. Returns the new mob_holder object.
+  */
+/mob/living/proc/mob_to_item()
 	var/obj/item/clothing/head/mob_holder/holder = new(get_turf(src), src, held_state, head_icon, held_lh, held_rh, worn_slot_flags)
-	L.visible_message("<span class='warning'>[L] scoops up [src]!</span>")
-	L.put_in_hands(holder)
+	return holder
+
+/**
+  * Creates a mob_holder for the mob and gives it to the picker-upper.
+  *
+  * Arguments:
+  * * user - mob currently picking up this mob
+  */
+/mob/living/proc/mob_pickup(mob/living/user)
+	var/obj/item/clothing/head/mob_holder/holder = mob_to_item()
+	user.visible_message("<span class='warning'>[user] scoops up [src]!</span>")
+	user.put_in_hands(holder)
```

Comment:
Not entirely sure what reason there is for this to be broken up into two procs

Reply:
Only needed that snippet of the code to implement into the storage obj code. Opted to make it a proc over copy+paste as any future atoms creating mob holders would also need that snippet.